wordpress-mobile/WordPress-Android
# App Data Coverage: Add more tracking to Reader Detail

Parent #16343 

This PR adds more tracking for the Reader Post Detail view:
- For all methods of ReaderWebViewUrlClickListener
- For featured image tapped
- Custom view shown/hidden

**Merge Instructions**
- Ensure the #16440 has been merged
- Remove the Not Ready for Merge label
- Merge as normal

**To test:**
- Launch app and login
- Prereq: Follow en.blog.wordpress.com & banterrepublic.blog (Reader Tab -> Manage -> Follow site)
--------------------------
- Navigate to Reader
- Navigate to the Following Tab (Use the filter to slim the list down if you are following a tons of blogs)
- Scroll to a WordPress.com (The one from April 20 works well - it has a featured image and links)
- Tap on the post card to launch the post detail view
- Tap the featured image within the article
- Verify log contains: 🔵 Tracked: reader_article_featured_image_tapped
- Close the image view
- Scroll to a link within the article and tap it
- Verify log contains: 🔵 Tracked: reader_article_link_tapped
- Scroll to a video within the article, tap to start it and then tap the full screen icon in the lower right hand corner
- Verify the log contains: 🔵 Tracked: reader_article_custom_view_shown
- Close the video view
- Verify the log contains: 🔵 Tracked: reader_article_custom_view_hidden
- Go back to the posts list 
- Scroll to a post that has an image link (I found on at banterrepublic.blog)
- Open the post
- Scroll to the image in the article and tap it
- Verify log contains: 🔵 Tracked: reader_article_image_tapped

## Regression Notes
1. Potential unintended areas of impact
Only added tracking

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
